core/trino-main/src/main/java/io/trino/memory/QueryContext.java

```diff
@@ -73,8 +72,6 @@
     private final SpillSpaceTracker spillSpaceTracker;
     private final Map<TaskId, TaskContext> taskContexts = new ConcurrentHashMap<>();
 
-    @GuardedBy("this")
-    private boolean resourceOverCommit;
     private volatile boolean memoryLimitsInitialized;
```

Comment:
nit: `memoryPool` can not be final. Also `memoryPoolMoveFuture` doesn't seem to be anymore needed.